### PR TITLE
fix: fix bug on protocol open command send by control

### PR DESCRIPTION
**Pull request by driftluo**
Fix bug on protocol open command send by control

### CODE REVIEW

File: src/service.rs [hunk 1]
```diff
@@ -970,7 +970,22 @@ where
     ) {
         if source == Source::External {
             debug!("try open session [{}] proto [{}]", id, proto_id);
-            if self.sessions.contains_key(&id) {
+            // The following 3 conditions must be met at the same time to send an event:
+            //
+            // 1. session must open
+            // 2. session protocol must does't open
```
Comment (doitian) [inline src/service.rs]:
```suggestion
            // 2. session protocol mustn't open
```

File: src/service.rs [hunk 2]
```diff
@@ -970,7 +970,22 @@ where
...
+            // 3. session protocol handle must doesn't exist
```
Comment (doitian) [inline src/service.rs]:
```suggestion
            // 3. session protocol handle mustn't exist
```

File: src/service.rs [hunk 3]
```diff
@@ -970,7 +970,22 @@ where
...
+            //
+            // Satisfy 2 but not satisfied 3 may cause an error, causing the service's session handle sender
```
Comment (doitian) [inline src/service.rs]:
```suggestion
            // Satisfy 2 but not 3 may cause an error, leading to the service's session handle sender
```

File: src/service.rs [hunk 4]
```diff
@@ -984,8 +999,8 @@ where
                     },
                 ));
                 self.distribute_to_session();
-                return;
             }
+            return;
```
Comment (doitian) [inline src/service.rs]: Why move `return` here? Another bug?

Comment (driftluo) [reply src/service.rs]: yes, it should be returned immediately, whether send an event on or not.

**Pull request merged**